### PR TITLE
feat: collapse side panels

### PR DESCRIPTION
**Pull request by aniruddha-mithya**
This will fix [issue#6](https://github.com/comake/rml-mapping-editor/issues/6).

Collapsed panels can be opened by clicking on their respective names that appear on the top right after they've been collapsed.

### CODE REVIEW

File: src/components/DraggableViewContainer.tsx [hunk 1]
```diff
@@ -11,114 +18,217 @@ export interface DraggableViewContainerProps<T extends number> {
...
+      // Space occupied by items that were previously collapsed and have now been opened
+      let uncollapsedItemsSpace = 0;
+      let openItemsCount = 0;
+      dimensions.forEach((dim, index) => {
+        if (collapsedIndices.includes(index) && dim !== undefined && dim > 0) {
+          collapsedItemsSpace += dim;
+        } else if (
+          !collapsedIndices.includes(index) &&
+          dim !== undefined &&
+          dim === 0
```
Comment (adlerfaulkner) [inline src/components/DraggableViewContainer.tsx:82]: Can just use `dim === 0` because `undefined !== 0`

File: src/components/Header.tsx [hunk 2]
```diff
@@ -1,13 +1,29 @@
...
     <div className={styles.header}>
       <div className={styles.logo}>RML Mapping Editor</div>
+      {collapsedItems.map((item) => (
+        <button
+          key={item}
+          onClick={unCollapse.bind(null, item)}
+          className={`${styles.panelHeaderButton}`}
+        >
+          {item[0].toUpperCase()}
+          {item.slice(1)} Panel
```
Comment (adlerfaulkner) [inline src/components/Header.tsx:21]: I can replace this with an icon

**Comment by aniruddha-mithya** [comment]
Have updated the code. Excluding icons in the header, other comments have been addressed. Regarding icons, I have updated the header component to support icons. Will update once the icons are available.

**Comment by adlerfaulkner** [comment]
Closed on favor of working in one repo in #20

**Pull request closed without merging**